src/sysinfo.cc

```diff
@@ -75,11 +75,12 @@ bool ReadIntFromFile(const char* file, long* value) {
     char line[1024];
     char* err;
     memset(line, '\0', sizeof(line));
-    CHECK(read(fd, line, sizeof(line) - 1));
-    const long temp_value = strtol(line, &err, 10);
-    if (line[0] != '\0' && (*err == '\n' || *err == '\0')) {
-      *value = temp_value;
-      ret = true;
+    if (read(fd, line, sizeof(line) - 1) >= 0) {
```

Comment:
I have no problems in changing it to "> 0", but in case of an error `read` returns negative number as error. Non-negative number means that `read` succsesfully finished. The case when nothing is read is covered on line 80.